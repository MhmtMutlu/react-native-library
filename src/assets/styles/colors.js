const colors = {
  grey: '#6c757d',
  darkGrey: '#495057',
  darkGrey2x: '#343a40',
  lightGrey: '#adb5bd',
  lightGrey2x: '#dee2e6',
  lightGrey3x: '#e0e0e0',
  white: '#f8f9fa',
};

export default colors;
