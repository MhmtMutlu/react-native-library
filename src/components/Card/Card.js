import React from 'react';
import {TouchableOpacity, Text, View} from 'react-native';

import {tr} from 'date-fns/locale';
import {formatDistance, parseISO} from 'date-fns';

import styles from './Card.styles';

const Card = ({cardData, index, onPress}) => {
  const formattedDate = date =>
    formatDistance(parseISO(date), new Date(), {
      addSuffix: true,
      locale: tr,
    });

  return (
    <TouchableOpacity style={styles.container} onPress={onPress}>
      <View style={styles.left_container}>
        <View style={styles.circle}>
          <Text style={styles.circle_text}>{index}</Text>
        </View>
      </View>
      <View style={styles.right_container}>
        <Text style={styles.title}>{cardData.bookName}</Text>
        <Text style={styles.writer}>{cardData.writer}</Text>
      </View>
    </TouchableOpacity>
  );
};

export default Card;
