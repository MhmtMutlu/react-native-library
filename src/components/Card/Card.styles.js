import {StyleSheet} from 'react-native';
import colors from '../../assets/styles/colors';

export default StyleSheet.create({
  container: {
    flex: 1,
    flexDirection: 'row',
    backgroundColor: colors.lightGrey3x,
    borderRadius: 5,
    borderTopLeftRadius: 50,
    borderBottomLeftRadius: 50,
    margin: 10,
    padding: 5,
    shadowColor: colors.grey,
    shadowOffset: {
      width: 0,
      height: 1,
    },
    shadowOpacity: 0.5,
    shadowRadius: 15,
    elevation: 10,
    height: 80,
  },
  left_container: {
    flex: 0.2,
    justifyContent: 'center',
    alignItems: 'center',
  },
  circle: {
    justifyContent: 'center',
    alignItems: 'center',
    color: colors.lightGrey3x,
    backgroundColor: colors.darkGrey2x,
    width: 80,
    height: 78,
    borderRadius: 50,
    borderWidth: 1,
    borderColor: colors.lightGrey3x,
  },
  circle_text: {
    color: colors.lightGrey3x,
    fontSize: 28,
    fontWeight: 'bold',
  },
  right_container: {
    flex: 0.8,
    justifyContent: 'center',
    alignItems: 'flex-start',
    paddingLeft: 20,
  },
  title: {
    fontSize: 24,
    fontWeight: 'bold',
    color: colors.darkGrey2x,
  },
  writer: {
    fontSize: 14,
    fontStyle: 'italic',
  },
});
