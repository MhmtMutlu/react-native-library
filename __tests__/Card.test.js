import React from 'react';
import {render} from '@testing-library/react-native';
import Card from '../src/components/Card';
import styles from '../src/components/Card/Card.styles';

const dummyData = {
  bookName: 'test book',
  writer: 'test writer',
};

const dummyIndex = 1;

test('should match with snapshot', () => {
  const comp = render(<Card cardData={dummyData} index={dummyIndex} />);
  expect(comp).toMatchSnapshot();
});

test('should render Card css', () => {
  const comp = render(<Card cardData={dummyData} index={dummyIndex} />);
  const card = comp.getByTestId('test-card').props.style;
  expect(card).toMatchSnapshot(styles.container);
});
