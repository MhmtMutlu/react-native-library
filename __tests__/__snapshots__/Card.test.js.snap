// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match with snapshot 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "#e0e0e0",
      "borderBottomLeftRadius": 50,
      "borderRadius": 5,
      "borderTopLeftRadius": 50,
      "elevation": 10,
      "flex": 1,
      "flexDirection": "row",
      "height": 80,
      "margin": 10,
      "opacity": 1,
      "padding": 5,
      "shadowColor": "#6c757d",
      "shadowOffset": Object {
        "height": 1,
        "width": 0,
      },
      "shadowOpacity": 0.5,
      "shadowRadius": 15,
    }
  }
  testID="test-card"
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 0.2,
        "justifyContent": "center",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#343a40",
          "borderColor": "#e0e0e0",
          "borderRadius": 50,
          "borderWidth": 1,
          "color": "#e0e0e0",
          "height": 78,
          "justifyContent": "center",
          "width": 80,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#e0e0e0",
            "fontSize": 28,
            "fontWeight": "bold",
          }
        }
      >
        1
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "flex-start",
        "flex": 0.8,
        "justifyContent": "center",
        "paddingLeft": 20,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#343a40",
          "fontSize": 24,
          "fontWeight": "bold",
        }
      }
    >
      test book
    </Text>
    <Text
      style={
        Object {
          "fontSize": 14,
          "fontStyle": "italic",
        }
      }
    >
      test writer
    </Text>
  </View>
</View>
`;

exports[`should render Card css 1`] = `
Object {
  "backgroundColor": "#e0e0e0",
  "borderBottomLeftRadius": 50,
  "borderRadius": 5,
  "borderTopLeftRadius": 50,
  "elevation": 10,
  "flex": 1,
  "flexDirection": "row",
  "height": 80,
  "margin": 10,
  "opacity": 1,
  "padding": 5,
  "shadowColor": "#6c757d",
  "shadowOffset": Object {
    "height": 1,
    "width": 0,
  },
  "shadowOpacity": 0.5,
  "shadowRadius": 15,
}
`;
